entrypoint.sh

```diff
@@ -1,136 +1,104 @@
     exec "${@}"
 fi
 
-if [ -f "$MUMBLE_CUSTOM_CONFIG_FILE" ]; then
-	# Just use the config file specified by the user and don't bother assembling our own
+if [[ -f "$MUMBLE_CUSTOM_CONFIG_FILE" ]]; then
 	echo "Using manually specified config file at $MUMBLE_CUSTOM_CONFIG_FILE"
 	echo "All MUMBLE_CONFIG variables will be ignored"
 	CONFIG_FILE="$MUMBLE_CUSTOM_CONFIG_FILE"
 else
-	# As a first step, we ensure that the config file is empty, so we can always start from a clean slate
 	echo -e "# Config file automatically generated from the MUMBLE_CONFIG_* environment variables\n" > "${CONFIG_FILE}"
 
-	used_configs=()
-	existing_config_options=()
-    
-	####
-	# Check what kind of configurations there exist in the bare bones config file
-	####
+	# Compile list of configurations that exist in bare bones config
+
 	while read -r line; do
 		if [[ "$line" =~ ^(\;|\#)?\ *([a-zA-Z_0-9]+)=.* ]]; then
 			existing_config_options+=("${BASH_REMATCH[2]}")
 		fi
 	done < "$BARE_BONES_CONFIG_FILE"
-	####
-	# Process settings following environments variables starting with "MUMBLE_CONFIG_"
-	# Iterate over all environment variable key, value pairs and check if they match our naming scheme
-	####
+
+	# Process settings through variables of format MUMBLE_CONFIG_*
+
 	while IFS='=' read -d '' -r var value; do
-		if [[ "$var" == "MUMBLE_CONFIG_"* ]]; then
-			uppercase_variable=${var/MUMBLE_CONFIG_/}
-
-			# Remove underscores (to ensure that it doesn't matter whether the user
-			# uses e.g. MUMBLE_CONFIG_DB_BLA or MUMBLE_CONFIG_DBBLA)
-			uppercase_variable_no_underscores="${uppercase_variable//_/}"
-			found=false
-
-			for current_config in "${existing_config_options[@]}"; do
-				# convert to uppercase
-				upper_current_config=${current_config^^}
-				
-				if [ "$upper_current_config" = "$uppercase_variable" ] || [ "$upper_current_config" = "$uppercase_variable_no_underscores" ]; then
-					set_config "$current_config" "$value"
-					found=true
-					break
-				fi
-			done
-
-			if [ "$found" = "false" ]; then
-				>&2 echo "[ERROR]: Unable to find config corresponding to variable \"$var\""
-				exit 1
+		uppercase_variable=${var/MUMBLE_CONFIG_/}
+		uppercase_variable_no_underscores="${uppercase_variable//_/}"
+		found=false
+
+		for current_config in "${existing_config_options[@]}"; do
+			upper_current_config=${current_config^^}
+
+			if [[ "$upper_current_config" = "$uppercase_variable" || "$upper_current_config" = "$uppercase_variable_no_underscores" ]]; then
+				set_config "$current_config" "$value"
+				found=true
+				break
 			fi
+		done
+
+		if [[ "$found" = "false" ]]; then
+			>&2 echo "[ERROR]: Unable to find config corresponding to variable \"$var\""
+			exit 1
 		fi
-	done < <( printenv --null ) # Feeding it in like this, prevents the creation of a subshell for the while-loop
+	done < <( printenv --null | grep -az MUMBLE_CONFIG_ ) # Feeding it in like this, prevents the creation of a subshell for the while-loop
```

Comment:
What does this do, if there is no `MUMBLE_CONFIG_*` variable defined? Will `grep` cause the script to error due to non-zero exit code?

Reply:
While `grep` will have a non-zero exit code, Bash's `set -e` won't break the script because of it. It's the exit code for the `while loop` that really matters, as far as I can tell, or at the very least the exit code for the command that generates the file descriptor seems not to matter for  `set -e`.

The script doesn't error out when a container is launched without any `MUMBLE_CONFIG_*` variables. A quick test shows Bash handles it well too:

```bash
#!/bin/bash
set -e

cat < <(printenv --null | grep -az MUMBLE_CONFIG_)
cat < <(false)
echo test
```

The script above outputs `test`.